connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3397,7 +3488,11 @@ class Dataset[T] private[sql] (
     sparkSession.analyze(plan, proto.AnalyzePlanRequest.AnalyzeCase.SCHEMA)
   }
 
-  def collectResult(): SparkResult[T] = sparkSession.execute(plan, agnosticEncoder)
+  def collectResult(): SparkResult[T] =
+    sparkSession.execute(plan, agnosticEncoder, getObservationsMapOpt)
+
+  def collectObservations(): Map[String, Map[String, Any]] =
+    collectResult().getObservedMetrics
```

Comment:
Where are we using this?

Reply:
So far only in tests.
Now this API is the only we we can fetch metrics that are not associated to `Observation` objects. It's serves the same propose as in the Python API, where we have `df.attrs["observed_metrics"]` to access such metrics.
Please give suggestion about a better place to place it.

Reply:
If it is only tests then drop it, and use `collectResult().getObservedMetrics` instead.

Reply:
Done!